src/Adapter/MSTest.CoreAdapter/ObjectModel/UnitTestElement.cs

```diff
@@ -109,20 +109,29 @@ public UnitTestElement(TestMethod testMethod)
         /// <returns> An instance of <see cref="TestCase"/>. </returns>
         internal TestCase ToTestCase()
         {
-            var fullName = string.Format(
-                CultureInfo.InvariantCulture,
-                "{0}.{1}",
-                this.TestMethod.FullClassName,
-                this.TestMethod.Name);
+            string fullName = this.TestMethod.HasManagedMethodAndType
+                            ? string.Format(CultureInfo.InvariantCulture, "{0}.{1}", this.TestMethod.ManagedTypeName, this.TestMethod.ManagedMethodName)
+                            : string.Format(CultureInfo.InvariantCulture, "{0}.{1}", this.TestMethod.FullClassName, this.TestMethod.Name);
 
             TestCase testCase = new TestCase(fullName, TestAdapter.Constants.ExecutorUri, this.TestMethod.AssemblyName);
+            if (string.IsNullOrWhiteSpace(this.DisplayName))
+            {
+                testCase.DisplayName = string.IsNullOrWhiteSpace(this.TestMethod.ManagedMethodName) ? this.TestMethod.Name : this.TestMethod.ManagedMethodName;
+            }
+            else
+            {
+                testCase.DisplayName = this.DisplayName;
+            }
 
-            testCase.DisplayName = string.IsNullOrEmpty(this.DisplayName) ? this.TestMethod.Name : this.DisplayName;
-            testCase.SetPropertyValue(TestAdapter.Constants.TestClassNameProperty, this.TestMethod.FullClassName);
             if (this.TestMethod.HasManagedMethodAndType)
             {
-                testCase.SetPropertyValue(TestCaseExtensions.ManagedTypeProperty, this.TestMethod.ManagedType);
-                testCase.SetPropertyValue(TestCaseExtensions.ManagedMethodProperty, this.TestMethod.ManagedMethod);
+                testCase.SetPropertyValue(TestCaseExtensions.ManagedTypeProperty, this.TestMethod.ManagedTypeName);
+                testCase.SetPropertyValue(TestCaseExtensions.ManagedMethodProperty, this.TestMethod.ManagedMethodName);
+                testCase.SetPropertyValue(TestAdapter.Constants.TestClassNameProperty, this.TestMethod.ManagedTypeName);
```

Comment:
nit: Could we put these properties in one place? They seem to be in `TestCaseExtensions` and `TestAdapter.Constants`.

Reply:
`TestCaseExtensions` extensions are coming from the `AdapterUtilities` NuGet, `TestClassNameProperty` is internal to TestFx.